test_runner/regress/test_remote_storage.py

```diff
@@ -673,35 +681,85 @@ def test_empty_branch_remote_storage_upload_on_restart(
     env = neon_env_builder.init_start()
     client = env.pageserver.http_client()
 
-    new_branch_name = "new_branch"
-    new_branch_timeline_id = env.neon_cli.create_branch(new_branch_name, "main", env.initial_tenant)
+    client.configure_failpoints(("before-upload-index", "return"))
 
-    with env.endpoints.create_start(new_branch_name, tenant_id=env.initial_tenant) as endpoint:
-        wait_for_last_flush_lsn(env, endpoint, env.initial_tenant, new_branch_timeline_id)
-    wait_upload_queue_empty(client, env.initial_tenant, new_branch_timeline_id)
```

Comment:
Add another `assert_upload_queue_empty` which could be last_record_lsn == remote_constitent_lsn.

Reply:
Done in [d8804b1](https://github.com/neondatabase/neon/pull/4204/commits/d8804b146525c91a89a4c716b35dbb6050900e05)